### PR TITLE
feat(exporter): Jaeger and Zipkin exporter use `logr` as the logging interface

### PR DESCRIPTION
**Pull request by aniaan**
close: #3497

Please help to see if this change is appropriate, cc @MadVikingGod

### CODE REVIEW

File: exporters/jaeger/uploader.go [hunk 1]
```diff
@@ -112,8 +114,18 @@ func WithAgentPort(port string) AgentEndpointOption {
...
 // WithLogger sets a logger to be used by agent client.
 func WithLogger(logger *log.Logger) AgentEndpointOption {
+	return agentEndpointOptionFunc(func(o agentEndpointConfig) agentEndpointConfig {
+		o.Logger = stdr.New(logger)
+		return o
+	})
+}
+
+// WithLogr sets a logr.Logger to be used by agent client.
+func WithLogr(logger logr.Logger) AgentEndpointOption {
```
Comment (Aneurysm9) [inline exporters/jaeger/uploader.go:128]:
```suggestion
func WithLogger(logger *log.Logger) AgentEndpointOption {
	return WithLogr(stdr.New(logger))
}

// WithLogr sets a logr.Logger to be used by agent client.
func WithLogr(logger logr.Logger) AgentEndpointOption {
```

I think I would implement one in terms of the other.

Comment (aniaan) [reply exporters/jaeger/uploader.go]: Yes, it's better this way, updated

File: exporters/zipkin/zipkin.go [hunk 2]
```diff
@@ -65,6 +69,11 @@ func (fn optionFunc) apply(cfg config) config {
 
 // WithLogger configures the exporter to use the passed logger.
 func WithLogger(logger *log.Logger) Option {
```
Comment (MadVikingGod) [inline exporters/zipkin/zipkin.go:71]: We should have some comments that `WithLogger` and `WithLogr` will overwrite each other.

File: exporters/jaeger/uploader.go [hunk 3]
```diff
@@ -112,8 +115,15 @@ func WithAgentPort(port string) AgentEndpointOption {
 	})
 }
 
+var emptyLogger = logr.Logger{}
+
 // WithLogger sets a logger to be used by agent client.
 func WithLogger(logger *log.Logger) AgentEndpointOption {
```
Comment (MadVikingGod) [inline exporters/jaeger/uploader.go:121]: We should have some comments that `WithLogger` and `WithLogr` will overwrite each other.

**Review by MadVikingGod — APPROVED** [review]
LGTM,

File: exporters/zipkin/zipkin.go [hunk 4]
```diff
@@ -36,20 +39,21 @@ const (
...
+	logger logr.Logger
 
 	stoppedMu sync.RWMutex
 	stopped   bool
 }
 
 var (
-	_ sdktrace.SpanExporter = &Exporter{}
+	_           sdktrace.SpanExporter = &Exporter{}
+	emptyLogger                       = logr.Logger{}
```
Comment (MrAlias) [inline exporters/zipkin/zipkin.go:50]: Don't mix type assertions with var declarations, it becomes hard to read and distinguish.

```suggestion
	_ sdktrace.SpanExporter = &Exporter{}
)

var (
	emptyLogger = logr.Logger{}
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -40,6 +40,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `Temporality(view.InstrumentKind) metricdata.Temporality` and `Aggregation(view.InstrumentKind) aggregation.Aggregation` methods are added to the `"go.opentelemetry.io/otel/exporters/otlp/otlpmetric".Client` interface. (#3260)
 - The `WithTemporalitySelector` and `WithAggregationSelector` `ReaderOption`s have been changed to `ManualReaderOption`s in the `go.opentelemetry.io/otel/sdk/metric` package. (#3260)
 - The periodic reader in the `go.opentelemetry.io/otel/sdk/metric` package now uses the temporality and aggregation selectors from its configured exporter instead of accepting them as options. (#3260)
+- Jaeger and Zipkin exporter use `github.com/go-logr/logr` as the logging interface, and add the `WithLogr` option. (#3497)
```
[the remark this answers is not in the document]
Comment (aniaan) [reply CHANGELOG.md]: updated

File: exporters/zipkin/zipkin.go (hunk 4 above)
Comment (aniaan) [reply exporters/zipkin/zipkin.go]: updated

**Review by MrAlias — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**